### PR TITLE
smaller docker image (2 stage build)

### PR DESCRIPTION
**Pull request by WuTheFWasThat**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ jobs:
 
   - stage: deploy
     install: skip
+    before_install:
+      - sudo apt-get update
+      - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
```
Comment (willprice) [inline .travis.yml]: I presume we need this to get 17.06+ for multistage builds?

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,35 @@
-FROM node:6-stretch
+FROM node:6-stretch AS build
 LABEL maintainer="will.price94@gmail.com"
 LABEL version="0.0.1"
-RUN apt-get update -qq && \
-    apt-get install -y python
 # Prevent npm from spamming
+ENV NPM_CONFIG_LOGLEVEL=warn
```
Comment (willprice) [inline Dockerfile]: Why? Do the lines below not achieve this, if not can we get rid of them?

File: Dockerfile [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
 RUN npm config set loglevel=warn && \
     npm config set progress=false
 WORKDIR /app/
 COPY package.json package-lock.json ./
 RUN npm install
 COPY . .
-RUN npm run build
+RUN mkdir -p /build/client
```
Comment (willprice) [inline Dockerfile]: How about `mkdir -p /build/{client,server}` and drop the next mkdir

File: Dockerfile [hunk 4]
```diff
@@ -1,24 +1,35 @@
...
+RUN npm run build -- --outdir /build/client
+RUN mkdir -p /build/server
+RUN npm run buildserver -- --outdir /build/server
 
+FROM node:6-stretch
```
Comment (willprice) [inline Dockerfile]: I wonder whether we can get away with [`node:6-slim`](https://github.com/nodejs/docker-node/blob/17c50cb300581280805a4183524fbf57840f3a7e/6.11/slim/Dockerfile)

Comment (willprice) [reply Dockerfile]: Yes, yes we can! We save 750MB!

Comment (willprice) [reply Dockerfile]: We could possibly make it even smaller by using alpine, but I worry that the `sqlite` stuff won't work as it uses `node-gyp` and if we compile on debian then move over to alpine we may well see glibc/musl issues.

Comment (willprice) [reply Dockerfile]: Nope, not an issue, Woo hoo!

File: .travis.yml (hunk 1 above)
Comment (WuTheFWasThat) [reply .travis.yml]: sounds about right.  not sure about the specific version, but travis was using an old version that couldn't parse the "FROM blah AS blah" format

File: Dockerfile (hunk 2 above)
Comment (WuTheFWasThat) [reply Dockerfile]: yeah `npm config set loglevel` didnt seem to work for me (not sure why).  removed it

File: Dockerfile (hunk 3 above)
Comment (WuTheFWasThat) [reply Dockerfile]: done

**Pull request merged**